### DISCUSSION TITLE
React - children height is larger then parent height

### DISCUSSION THREAD

**Discussion by obkim-hello** [tailwindlabs/tailwindcss#9478] [Help]
![Screen Shot 2022-10-04 at 12 37 10](https://user-images.githubusercontent.com/60444524/193876235-3b5369d2-ef64-4fce-9972-9d353a99153e.png)

I have no "h-screen" in all the child component below the "h-full" in the screenshot.

my react project is have a structure like 
![Screen Shot 2022-10-04 at 12 50 45](https://user-images.githubusercontent.com/60444524/193879384-a4ec557a-26e2-424a-8a1e-9b94a4841aff.png)

![Screen Shot 2022-10-04 at 12 51 04](https://user-images.githubusercontent.com/60444524/193879436-d20fb6e6-44cd-4b85-97ad-462cfee3f06e.png)

The problem:

1. everything in {children} does not follow the parent(the < Layout />) height
2. every page that i used < Layout /> having the same issue

**Comment by wongjn** [comment]
For `height: 100%` to work, the parent of the element that has `height: 100%` must also have an explicit height (there are other situations as well, such as grid or flex layout, but those are not relevant in your current situation). Your `div.bg-main` element that is the child of the element you have highlighted in devtools in your screenshot does not appear to have an explicit height, so in essence, `height: 100%` does nothing here. You could look at adding `height: 100%` with `h-full` or `height: 100vh` with `h-screen` to this `div.bg-main` if you want to be able to set children's heights as percentage of the parent.

**Reply by obkim-hello** [reply]
![Screen Shot 2022-10-04 at 13 05 23](https://user-images.githubusercontent.com/60444524/193882209-7276a9e4-fa5c-4a96-8567-4579d85a3c3e.png)

Is this what you mean?

**Reply by obkim-hello** [reply]
Isn't that all child's height should follow the parent height when the child's height is not defined?

**Reply by wongjn** [reply]
No, a child's height will be the `min-content` height, which in most cases, is the "natural" height of its content.
